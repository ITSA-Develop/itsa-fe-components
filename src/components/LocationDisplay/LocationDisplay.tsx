import { getMappedOptionFromAPI } from '@/helpers';
import { TInputOptions } from '@/types';
import { InfoRow } from './components/InfoRow/InfoRow';

interface ILocationDisplayProps {
	isEcuador: boolean;
	label: string;
	canton?: number;
	province?: number;
	country?: number;
	parishOptions?: TInputOptions[];
	cantonOptions?: TInputOptions[];
	provinceOptions?: TInputOptions[];
	countryOptions: TInputOptions[];
}

export const LocationDisplay = ({
	isEcuador,
	label,
	canton,
	province,
	country,
	cantonOptions = [],
	provinceOptions = [],
	countryOptions,
}: ILocationDisplayProps) => {
	const locationLabel = isEcuador
		? `${getMappedOptionFromAPI(canton, cantonOptions).label}, ${getMappedOptionFromAPI(province, provinceOptions).label}, ${getMappedOptionFromAPI(country, countryOptions).label}`
		: getMappedOptionFromAPI(country, countryOptions).label;

	return <InfoRow label={label} value={locationLabel} />;
};
