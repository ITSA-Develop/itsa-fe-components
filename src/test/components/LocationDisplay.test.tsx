import { render, screen } from '@testing-library/react';
import { LocationDisplay } from '../../components/LocationDisplay/LocationDisplay';

const mockOptions = [
	{ label: 'Option 1', value: 1 },
	{ label: 'Option 2', value: 2 },
	{ label: 'Option 3', value: 3 },
];

vi.mock('@/helpers', () => ({
	getMappedOptionFromAPI: (value: number | undefined, options: any[]) =>
		options.find(opt => opt.value === value) || { label: '' },
}));

describe('LocationDisplay', () => {
	it('shows concatenated label when isEcuador is true', () => {
		const { container } = render(
			<LocationDisplay
				isEcuador={true}
				label="Ubicación"
				canton={1}
				province={2}
				country={3}
				cantonOptions={mockOptions}
				provinceOptions={mockOptions}
				countryOptions={mockOptions}
			/>,
		);

		expect(screen.getByText('Ubicación')).toBeInTheDocument();
		expect(screen.getByText('Option 1, Option 2, Option 3')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('shows only country when isEcuador is false', () => {
		const { container } = render(
			<LocationDisplay isEcuador={false} label="País" country={2} countryOptions={mockOptions} />,
		);

		expect(screen.getByText('País')).toBeInTheDocument();
		expect(screen.getByText('Option 2')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('shows only country when isEcuador is false', () => {
		const { container } = render(
			<LocationDisplay
				isEcuador={true}
				label="Ubicación"
				canton={99}
				province={99}
				country={99}
				cantonOptions={mockOptions}
				provinceOptions={mockOptions}
				countryOptions={mockOptions}
			/>,
		);

		expect(screen.getByText('Ubicación')).toBeInTheDocument();

		const valueElement = screen.getByText((content, element) => {
			return element?.tagName.toLowerCase() === 'span' && content.includes(', ,');
		});
		expect(valueElement).toHaveTextContent(', ,');
		expect(container).toMatchSnapshot();
	});
});
