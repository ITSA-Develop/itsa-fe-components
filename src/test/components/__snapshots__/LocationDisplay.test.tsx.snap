// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LocationDisplay > shows concatenated label when isEcuador is true 1`] = `
<div>
  <div
    class="flex justify-between gap-2"
  >
    <span
      class="font-semibold"
    >
      Ubicación
    </span>
    <span
      class="text-right"
    >
      Option 1, Option 2, Option 3
    </span>
  </div>
</div>
`;

exports[`LocationDisplay > shows only country when isEcuador is false 1`] = `
<div>
  <div
    class="flex justify-between gap-2"
  >
    <span
      class="font-semibold"
    >
      País
    </span>
    <span
      class="text-right"
    >
      Option 2
    </span>
  </div>
</div>
`;

exports[`LocationDisplay > shows only country when isEcuador is false 2`] = `
<div>
  <div
    class="flex justify-between gap-2"
  >
    <span
      class="font-semibold"
    >
      Ubicación
    </span>
    <span
      class="text-right"
    >
      , , 
    </span>
  </div>
</div>
`;
